libraries/ms-common/inc/spi.h

```diff
@@ -0,0 +1,111 @@
+// SPI HAL Interface
+
+#include "interrupt.h"
+#include "status.h"
+#include "stm32f0xx.h"
```

Comment:
I don't think you actually need this header here. The ``stm32f0xx.h`` header contains STM32F0-specific stuff, which probably doesn't need to be exposed in the interface.